### PR TITLE
Remove sui popup

### PR DESCRIPTION
**Pull request by MiZiet**

### CODE REVIEW

File: packages/react-components/src/Popup/PopupWindow.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+import React from 'react';
+import { createPortal } from 'react-dom';
+import styled, { css } from 'styled-components';
+
+import { useElementPosition } from '@polkadot/react-hooks';
+
+import { getPosition } from './utils';
+
+export const POINTER_OFFSET = 14 * 0.8;
```
Comment (wachulski) [inline packages/react-components/src/Popup/PopupWindow.tsx:14]: Could you put a line of docs - why such values?

File: packages/react-components/src/Popup.tsx [hunk 2]
```diff
@@ -1,34 +1,122 @@
...
+
+interface WindowProps {
+  position: 'left' | 'middle' | 'right'
+}
+
+const PopupWindow = styled.div<WindowProps>`
+  position: absolute;
+  margin-top: 0.7rem;
+  top: 100%;
+  //bottom: 0;
```
Comment (wachulski) [inline packages/react-components/src/Popup.tsx]: not needed?

**Review by wachulski — APPROVED** [review]
LGTM

File: packages/react-components/src/Popup/PopupWindow.tsx (hunk 1 above)
Comment (MiZiet) [reply packages/react-components/src/Popup/PopupWindow.tsx]: ✅

File: packages/react-components/src/Popup.tsx (hunk 2 above)
Comment (MiZiet) [reply packages/react-components/src/Popup.tsx]: I think it's not in the code anymore, you must have looked at some old commit

File: packages/react-components/src/Popup/PopupWindow.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+import React from 'react';
+import { createPortal } from 'react-dom';
+import styled, { css } from 'styled-components';
+
+import { useElementPosition } from '@polkadot/react-hooks';
+
+import { getPosition } from './utils';
+
+export const PopupWindow: React.FC<PopupWindowProps> = ({ children, position, setRef, triggerPosition }) => {
```
Comment (jacogr) [inline packages/react-components/src/Popup/PopupWindow.tsx:14]:
```suggestion
function PopupWindow({ children, position, setRef, triggerPosition }: PopupWindowProps) {
```

File: packages/react-components/src/Popup/PopupWindow.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      ref={setRef}
+      windowPosition={getPosition(triggerPosition, position, triggerPosition.verticalPosition, windowPosition)}
+    >
+      {children}
+    </Window>,
+    document.body
+  );
+};
+
+const Window = styled.div<WindowProps>`
```
Comment (jacogr) [inline packages/react-components/src/Popup/PopupWindow.tsx:30]: Don't use this form in styled-components, rather follow the same pattern as used elsewhere, i.e. div + className + styles on React.memo

File: packages/react-components/src/Popup/index.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+          />
+        )}
+      </div>
+    </PopupWrapper>
+  );
+};
+
+export default React.memo(Popup);
+
+const PopupWrapper = styled.div`
```
Comment (jacogr) [inline packages/react-components/src/Popup/index.tsx:58]: Include styles alongside React.memo

File: packages/react-components/src/styles/semantic.ts [hunk 6]
```diff
@@ -171,23 +171,6 @@ export default (_theme: ThemeDef): string => `
     display: flex !important;
   }
 
-  .ui.popup {
```
Comment (jacogr) [inline packages/react-components/src/styles/semantic.ts:174]: +1

File: packages/react-hooks/src/useElementPosition.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  height: number,
+  verticalPosition: 'bottom' | 'top'
+}
+
+export function useElementPosition (ref: React.MutableRefObject<HTMLElement | undefined | null>): ElementPosition | undefined {
+  const [elementPosition, setElementPosition] = useState<ElementPosition>();
+
+  const windowSize = useWindowSize();
+
+  const scrollY = useScroll();
```
Comment (jacogr) [inline packages/react-hooks/src/useElementPosition.ts:24]:
```suggestion
  const [elementPosition, setElementPosition] = useState<ElementPosition>();
  const windowSize = useWindowSize();
  const scrollY = useScroll();
```

File: packages/react-hooks/src/useScroll.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect, useState } from 'react';
+
+export function useScroll (): number {
+  const [scrollY, setScrollY] = useState(0);
+
+  function logit () {
+    setScrollY(window.pageYOffset);
+  }
```
Comment (jacogr) [inline packages/react-hooks/src/useScroll.ts:11]: useCallback

**Review by jacogr — COMMENTED** [review]
It is great having these component replaced. Some small comments.

File: packages/react-hooks/src/useScroll.ts (hunk 8 above)
Comment (MiZiet) [reply packages/react-hooks/src/useScroll.ts]: ✅

File: packages/react-hooks/src/useElementPosition.ts (hunk 7 above)
Comment (MiZiet) [reply packages/react-hooks/src/useElementPosition.ts]: ✅

File: packages/react-components/src/Popup/index.tsx (hunk 5 above)
Comment (MiZiet) [reply packages/react-components/src/Popup/index.tsx]: ✅

File: packages/react-components/src/Popup/PopupWindow.tsx (hunk 4 above)
Comment (MiZiet) [reply packages/react-components/src/Popup/PopupWindow.tsx]: ✅

File: packages/react-components/src/Popup/PopupWindow.tsx (hunk 3 above)
Comment (MiZiet) [reply packages/react-components/src/Popup/PopupWindow.tsx]: ✅

**Review by jacogr — COMMENTED** [review]
The placing here is really funny at points. Sometimes it works, sometimes it doesn't seem to. See below.

image

Wondering if a horizonal arrow and placing it to the left is not more sane. Instead of trying to move it above/below, simplify?

Additionally would make the triangle slightly larger.

From a style perspective, it is still just as ugly (don't take this as a criticism) as the old SUI version, but after this we have better control so can make it fit in better, which is absolutely great.

**Comment by jacogr** [comment]
Yes, the calcs are indeed an issue, hence the question if we maybe won't be better suited moving it horizontally to the left. (Not a request, rather a suggestion _if_ it makes it easier) Will play through now.

**Review by jacogr — APPROVED** [review]
lgtm.

**Pull request merged**